{"session_id": "devin-5d864375992d4cb7a16cc7259e2e432f", "issues": [{"file": "src/waitress/channel.py", "line": 178, "description": "Potential NoneType error: self.request.expect_continue is accessed without checking if self.request is None"}, {"file": "src/waitress/channel.py", "line": 187, "description": "Potential NoneType error: self.request.completed is accessed without checking if self.request is None"}, {"file": "src/waitress/channel.py", "line": 363, "description": "Type mismatch: ReadOnlyFileBasedBuffer is appended to outbufs which expects OverflowableBuffer objects"}, {"file": "src/waitress/channel.py", "line": 379, "description": "Potential NoneType error: _flush_exception may return None if flush is None, but the return value is unpacked"}, {"file": "src/waitress/channel.py", "line": 399, "description": "Potential NoneType error: _flush_exception may return None if flush is None, but the return value is unpacked"}, {"file": "src/waitress/channel.py", "line": 447, "description": "Attribute error: Cannot assign to attribute 'error' for class 'parser_class'"}, {"file": "src/waitress/task.py", "line": 167, "description": "Missing method: Task.execute() is called but not defined in the Task class"}, {"file": "src/waitress/trigger.py", "line": 89, "description": "Missing method: _triggerbase.del_channel() is called but not defined in the _triggerbase class"}, {"file": "src/waitress/trigger.py", "line": 90, "description": "Missing method: _triggerbase._close() is called but not defined in the _triggerbase class"}, {"file": "src/waitress/trigger.py", "line": 96, "description": "Missing method: _triggerbase._physical_pull() is called but not defined in the _triggerbase class"}, {"file": "src/waitress/trigger.py", "line": 100, "description": "Missing method: _triggerbase.recv() is called but not defined in the _triggerbase class"}, {"file": "src/waitress/trigger.py", "line": 109, "description": "Missing method: _triggerbase.log_info() is called but not defined in the _triggerbase class"}, {"file": "src/waitress/wasyncore.py", "line": 78, "description": "Undefined variable: socket_map is used before definition"}, {"file": "src/waitress/wasyncore.py", "line": 331, "description": "Type mismatch: sock.setblocking(0) passes integer 0 instead of boolean False"}, {"file": "src/waitress/wasyncore.py", "line": 392, "description": "Potential NoneType error: self.socket.send() is called without checking if self.socket is None"}, {"file": "src/waitress/wasyncore.py", "line": 406, "description": "Potential NoneType error: self.socket.recv() is called without checking if self.socket is None"}, {"file": "src/waitress/wasyncore.py", "line": 464, "description": "Potential NoneType error: self.socket.getsockopt() is called without checking if self.socket is None"}]}